File: iosApp/iosApp/Components/ResultTemplate.swift
import SwiftUI

/// The PWA's success/error result screens as one template (U18, KTD-11, R12):
/// centered 80pt circle (badge + check on success, danger/15 + X on failure),
/// headline, optional detail, the load-bearing "Your funds are safe."
/// reassurance on failures, and a full-width CTA. Copy is carried verbatim
/// from the PWA at the call sites (U19–U22).
struct ResultTemplate<Extra: View>: View {
    let success: Bool
    let headline: String
    let onCta: () -> Void
    var detail: String?
    var fundsAreSafe: Bool
    var ctaLabel: String = "Done"
    @ViewBuilder var extraContent: Extra

    @Environment(\.zinqqColors) private var colors

    var body: some View {
        VStack(spacing: 0) {
            Spacer()
            ZStack {
                Circle()
                    .fill(success ? colors.badge : colors.danger.opacity(0.15))
                    .frame(width: 80, height: 80)
                Image(systemName: success ? "checkmark" : "xmark")
                    .font(.system(size: 36, weight: .semibold))
                    .foregroundColor(success ? colors.onBadge : colors.danger)
            }
            .accessibilityLabel(success ? "Success" : "Failure")

            Text(headline)
                .font(ZinqqFont.display(success ? 34 : 24, weight: .bold))
                .foregroundColor(colors.onDark)
                .multilineTextAlignment(.center)
                .padding(.top, 24)

            if let detail {
                Text(detail)
                    .font(ZinqqFont.sans(14))
                    .foregroundColor(success ? colors.onDarkMuted : colors.danger)
                    .multilineTextAlignment(.center)
                    .padding(.top, 8)
            }

            if fundsAreSafe {
                Text("Your funds are safe.")
                    .font(ZinqqFont.sans(14))
                    .foregroundColor(colors.onDarkMuted)
                    .multilineTextAlignment(.center)
                    .padding(.top, 4)
            }

            if Extra.self != EmptyView.self {
                extraContent
                    .padding(.top, 16)
            }

            Button(action: onCta) {
                Text(ctaLabel)
                    .font(ZinqqFont.display(18, weight: .bold))
                    .foregroundColor(colors.onCta)
                    .frame(maxWidth: 280)
                    .frame(height: 56)
                    .background(colors.cta)
                    .clipShape(RoundedRectangle(cornerRadius: 12))
            }
            .padding(.top, 32)
            .accessibilityLabel(ctaLabel)
            Spacer()
        }
        .padding(.horizontal, 32)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(colors.dark.ignoresSafeArea())
    }
}

extension ResultTemplate where Extra == EmptyView {
    init(
        success: Bool,
        headline: String,
        onCta: @escaping () -> Void,
        detail: String? = nil,
        fundsAreSafe: Bool? = nil,
        ctaLabel: String = "Done"
    ) {
        self.init(
            success: success,
            headline: headline,
            onCta: onCta,
            detail: detail,
            fundsAreSafe: fundsAreSafe ?? !success,
            ctaLabel: ctaLabel
        ) { EmptyView() }
    }
}

File: androidApp/src/main/kotlin/zinqq/app/components/ResultTemplate.kt
package zinqq.app.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import zinqq.app.R
import zinqq.app.theme.ZinqqTheme

/**
 * The PWA's success/error result screens as one template (U13, KTD-11, R12):
 * centered 80dp circle (badge + check on success, danger/15 + X on failure),
 * headline, optional detail, the load-bearing "Your funds are safe."
 * reassurance on failures, and a full-width CTA. Copy is carried verbatim
 * from the PWA at the call sites (U14–U17).
 */
@Composable
fun ResultTemplate(
    success: Boolean,
    headline: String,
    onCta: () -> Unit,
    modifier: Modifier = Modifier,
    detail: String? = null,
    fundsAreSafe: Boolean = !success,
    ctaLabel: String = "Done",
    extraContent: (@Composable () -> Unit)? = null,
) {
    val colors = ZinqqTheme.colors
    Column(
        modifier = modifier
            .fillMaxSize()
            .background(colors.dark)
            .padding(horizontal = 32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Box(
            modifier = Modifier
                .size(80.dp)
                .clip(CircleShape)
                .background(if (success) colors.badge else colors.danger.copy(alpha = 0.15f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                painter = painterResource(if (success) R.drawable.ic_check else R.drawable.ic_x_close),
                contentDescription = if (success) "Success" else "Failure",
                tint = if (success) colors.onBadge else colors.danger,
                modifier = Modifier.size(40.dp),
            )
        }
        Text(
            text = headline,
            color = colors.onDark,
            fontFamily = ZinqqTheme.fonts.display,
            fontWeight = FontWeight.Bold,
            fontSize = if (success) 34.sp else 24.sp,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(top = 24.dp),
        )
        if (detail != null) {
            Text(
                text = detail,
                color = if (success) colors.onDarkMuted else colors.danger,
                fontSize = 14.sp,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(top = 8.dp),
            )
        }
        if (fundsAreSafe) {
            Text(
                text = "Your funds are safe.",
                color = colors.onDarkMuted,
                fontSize = 14.sp,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(top = 4.dp),
            )
        }
        extraContent?.let {
            Box(modifier = Modifier.padding(top = 16.dp)) { it() }
        }
        Box(
            modifier = Modifier
                .padding(top = 32.dp)
                .fillMaxWidth()
                .widthIn(max = 280.dp)
                .height(56.dp)
                .clip(RoundedCornerShape(12.dp))
                .background(colors.cta)
                .clickable(onClick = onCta),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                text = ctaLabel,
                color = colors.onCta,
                fontFamily = ZinqqTheme.fonts.display,
                fontWeight = FontWeight.Bold,
                fontSize = 18.sp,
            )
        }
    }
}
